utils/geoutils.py

```diff
@@ -180,4 +181,4 @@ def get_key_recursive(key, config):
     if isinstance(val, (dict, collections.OrderedDict)):
         assert len(key) > 1, "missing keys to index metadata subdictionaries"
         return get_key_recursive(key[1:], val)
-    return val
+    return int(val)
```

Comment:
Isn't this value already casted to int here: https://github.com/NRCan/geo-deep-learning/blob/develop/utils/geoutils.py#L31? 
If casting to int is necessary here, I would suggest doing this small modification [in the list-ids function at line 25](at https://github.com/NRCan/geo-deep-learning/blob/develop/utils/geoutils.py#L25), rather that in the generic get_key_recursive. If, in the future, this utility funciton is used to return anything else but an int, `int(val)` could raise an undesired exception. I'd keep `return val` here. Up to you if you want to integrate it or not.

Reply:
I would skip this change for now to merge. I would note this and fix in a subsequent PR